cmd/kubeadm/app/master/selfhosted_etcd.go

```diff
@@ -164,6 +173,69 @@ func CreateSelfHostedEtcdCluster(cfg *kubeadmapi.MasterConfiguration, client *cl
 	return nil
 }
 
+func createTLSAssets(client *clientset.Clientset) error {
+	if err := createTLSSecret(client, operatorSecret, "etcd", []string{}); err != nil {
+		return err
+	}
+
+	peerDNSNames := []string{
+		fmt.Sprintf("*.%s.%s.svc.cluster.local", etcdCluster, metav1.NamespaceSystem),
+	}
+	if err := createTLSSecret(client, peerSecret, "peer", peerDNSNames); err != nil {
+		return err
+	}
+
+	clientDNSNames := []string{
+		fmt.Sprintf("*.%s.%s.svc.cluster.local", etcdCluster, metav1.NamespaceSystem),
+		fmt.Sprintf("*.%s-client.%s.svc.cluster.local", etcdCluster, metav1.NamespaceSystem),
```

Comment:
Generate S1: CA1, Cert1, Key1
Generate S2: CA2, Cert2, Key2

CA2 + Cert1 + Key1 => etcd client port's secret
CA1 + Cert2 + Key2 => etcd client's secret

You need to swap them.

> I updated to v0.3.2 and it's not listening on HTTPS. Logs are here.

Can you try to reproduce it in with pure etcd operator and create issue?